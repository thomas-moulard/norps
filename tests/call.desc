Teste les appels de fonction.
