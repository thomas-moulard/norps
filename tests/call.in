open libc.so as c

libs
c::printf("toto\ttata\n");
(void) c::printf("tutu %i\n", 22);

(void) c::printf("aaaa %i\n", 45);

(int) c::strlen("toto\n");
(uint) c::strlen("toto\n");
(short) c::strlen("toto\n");
(ushort) c::strlen("toto\n");
(long) c::strlen("toto\n");
(ulong) c::strlen("toto\n");

(void) c::printf("he!");
(void) c::malloc(42);
(void) c::printf("he!");

close c

exit