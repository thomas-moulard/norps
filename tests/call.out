~ libc is opened as c

(void)

(void)
5
5
5
5
5
5

(void)

(void)

(void)
toto	tata
tutu 22
aaaa 45
he!he!